pkg/server/claims.go

```diff
@@ -157,6 +159,18 @@ func parsePrivateClaims(oidcToken jwt.Token) (*oidcClaims, error) {
 	return &claims, nil
 }
 
+func extractRepository(oidcToken jwt.Token) (string, error) {
+	if oidcToken.Issuer() == config.GitHubIssuer {
+		return requiredClaim(oidcToken, "repository")
+	}
+
+	if len(oidcToken.Audience()) != 1 {
+		return "", fmt.Errorf("non-github OIDC token's audience field should have exactly one entry of a repository containing a minty config")
+	}
+	scopeRepository, _ := strings.CutPrefix(oidcToken.Audience()[0], "https://github.com/")
```

Comment:
So, my understanding is that GitHub issued OIDC tokens have the `repository` claim and we just read the repo from that field. But non-GitHub issued tokens (e.g. one issued by a GCP service account) do not have that claim and include very minimal metadata, but when you generate the token you can set the `aud` field to anything (at least in GCP). So we will expect the audience field to point to the repository where the relevant minty config lives, either in `org/repo` format or `https://github.com/org/repo` format. This line just normalizes the audience to `org/repo` format. 

The design doc (go/ghtm-v2-design) actually mentions using the audience field for this purpose. 
> OIDC tokens that are not sourced from GitHub will use the aud (audience) field to denote the scope of the request. The scope denotes where to find the configuration information that determines if a request for elevated privileges is valid. This will typically be something like abcxyz/infra-gcp or a full GitHub URL like https://github.com/abcxyz/infra-gcp.

Reply:
Yes, the aud field is the correct choice for non-github tokens. I was just wondering if we should normalized on that and ignore the `repository` field on the GitHub tokens but that's probably a more involved change than we need to do right now.

Thanks.